### ISSUE TITLE
Changes needed to Veterans Minisite

### ISSUE THREAD

**Issue by dustinWeaver** [topcoder-platform/community-app#799]
I need to make the following changes to the Veterans minisite:

1. Change the "Active Challenge" section to show active challenges from the veteran community

1. Change the "Browse all challenges" button to point to https://veterans.topcoder.com/challenges?communityId=veterans

Screenshot for reference:

![topcoder](https://user-images.githubusercontent.com/1737883/37792718-8e50f7aa-2dda-11e8-83fd-8c501e4d75cd.png)

**Comment by birdofpreyru** [comment]
@cwdcwd this is an easy one, I'll do in a bit

**Comment by birdofpreyru** [comment]
Hm... actually, now I recall that we kind of had this behavior before: to show only Veterans-related challenges in Veterans, but then, some times, there was no challenges, so Tony asked me to update to the current behavior. @dustinWeaver @cwdcwd probably we need to double-check with @ajefts whether we really wanna change it now.

**Comment by dustinWeaver** [comment]
You're correct @birdofpreyru.  Before we didn't have any challenges, so nothing would show.  But now we actually have two dedicated copilots and will have a steady stream of challenges.

**Comment by birdofpreyru** [comment]
@dustinWeaver can you elaborate a bit more, what you want to see in the challenge listing in Veterans? The link you asked for, hides all non-veterans challenges there, by default; but we have there the dedicated *Active Veterans Challenges* bucket and option in the sidebar menu anyway, so with this filter enabled it looks a bit redundand. Should we drop the bucket? Or, probably, keep the current link, or anything else?

**Comment by dustinWeaver** [comment]
If I had a wish, it would function like this:

- If there were active challenges in the Veterans group, it would show those on both the minisite page and the challenge list

- If no active challenges, just show challenges from our public challenges

Does this make sense?

**Comment by birdofpreyru** [comment]
@dustinWeaver 

While attempting to make this update I have recalled that we can't show the veterans challenges to the non-members of Veterans communitiy (API forbids that). Thus, in the active challenges block at Home I can show Veterans challenges only to members of the group, which does not make that much sense, IMHO, as the Home page is targeted to non-members, primarily, I believe? I guess, this was another reason to show public challenges there.

**Comment by dustinWeaver** [comment]
Hmm.  @alamora, here is the issue with showing the vet challenges on the public site.

@birdofpreyru, would there be a way to provide a box that has a count of the number of challenges for the vet community?  Even if we could show them, at least we could say "There are X challenges with a total of $X prize money available"?

**Comment by birdofpreyru** [comment]
@dustinWeaver nope. We have a way to show the count and total prize of active challenges belonging to a community; but it, also, can count only the challenges visible to the user (under the hood it still has to fetch all active challenges, and then reduce them to the necessary stats to show). For a workaround we need to add dedicated endpoint to the API, and for that you have to talk with Tony.
